### ISSUE TITLE
LongParameterList baseline doesn't recognize signature after parameter count changes

### ISSUE THREAD

**Issue by MohammedHamza0631** [detekt/detekt#8785]
## Expected Behavior
When a function/constructor with a baselined LongParameterList violation has its parameter count changed (either increased or decreased, but still above threshold), the baseline should continue to recognize it as the same entity and not flag it as a new violation.

> For example:
> - Constructor has 17 parameters (threshold: 10) - baselined
> - Developer makes changes: reduces to 15 parameters OR increases to 19 parameters (still above threshold of 10)
> - Baseline should recognize this as the same constructor that was already baselined

## Observed Behavior
Detekt flags the modified constructor as a new violation because the baseline signature no longer matches, regardless of whether parameters were added or removed.

The baseline stores the complete parameter signature:
```xml
<ID>LongParameterList:TestClass.kt$TestClass$( private val param1: String, private val param2: Int, private val param3: Boolean, private val param4: Double, private val param5: Float, private val param6: Long, private val param7: Byte, private val param8: Short, private val param9: Char, private val param10: String )</ID>
```
When parameters change from 10 to either 8 or 12, Detekt cannot find a matching baseline entry (since the signature now has different parameters) and treats it as new code, flagging it as a violation.

## Steps to Reproduce
1. Create a constructor with parameters exceeding the threshold:
```yml
// detekt.yml - set threshold to 6
complexity:
  LongParameterList:
    constructorThreshold: 6
```
```kotlin
// TestClass.kt - 10 parameters > threshold
class TestClass(
    private val param1: String,
    private val param2: Int,
    private val param3: Boolean,
    private val param4: Double,
    private val param5: Float,
    private val param6: Long,
    private val param7: Byte,
    private val param8: Short,
    private val param9: Char,
    private val param10: String
)
```
2. Run Detekt → LongParameterList violation flagged
3. Create baseline → violation suppressed
4. Remove 2 parameters (8 params, still > threshold):
```kotlin
// TestClass.kt - 8 params, still > threshold
class TestClass(
    private val param1: String,
    private val param2: Int,
    private val param3: Boolean,
    private val param4: Double,
    private val param5: Float,
    private val param6: Long,
    private val param7: Byte,
    private val param8: Short,
)
```
## Context
This issue defeats the purpose of creating a baseline for large codebases. When working with legacy code:

1. We baseline existing violations to prevent them from blocking CI
2. We need to make changes to these classes/functions (add new dependencies, refactor existing ones)
3. Each parameter change gets flagged as a new violation, forcing us to either:
    1. Fix it completely (not always feasible in large codebases)
    2. Suppress it manually with @Suppress annotations
    3. Update the baseline after each change

## Your Environment

* Version of detekt used: 1.23.8
* Version of Gradle used (if applicable): 8.13
* Gradle scan link: https://scans.gradle.com/s/lmrwoe25eefec
* Operating System and version: macOS Sequoia 15.6.1
* Link to your project (if it's a public repository): https://github.com/MohammedHamza0631/CineLink

**Comment by BraisGabin** [comment]
More context here: https://github.com/detekt/detekt/discussions/8763

**Comment by atulgpt** [comment]
For classes, we can just use the class name. But what about secondary constructors, like the ones below?
```kt
constructor(a: Int, b: Int) : this() // this is valid  
constructor(a: Int, b: Int, c: Int, d: Int) : this() // this is not 
```

Similarly for functions, using just the name would add baselines for all their overloads.

We can either just do this for class primary constructors or ignore the false baseline and do it for all

**Comment by BraisGabin** [comment]
I understand the problem. The baseline can't be perfect because the code doesn't have stable ids. So, to me, we should allow those.

If we really care about the problem of two issues with the same signature (probably we should) we could improve the baseline implementation. We could add duplicated signatures. Ig there are two signatures for the same rule and fike that means that we should allow 2 issues with the same signature. But we shouldn't allow 3. That makes the baseline implementation a bit more difficult but probably it is worth it.

**Comment by MohammedHamza0631** [comment]
So as per the solution proposed in #8816 are you trying to say this(this is my understanding, correct me if wrong):
Now the baseline stores only `1 entry per rule+file+violation(Set like behaviour)`, now if we do what you are saying then the baseline will store all the violations in that file for that rule.
 
But i can't see how this solves the issue listed above(#8785)?? Right now the baseline entry for this is:
```xml
<ID>LongParameterList:TestClass.kt$TestClass$( private val param1: String, private val param2: Int, private val param3: Boolean, private val param4: Double, private val param5: Float, private val param6: Long, private val param7: Byte, private val param8: Short, private val param9: Char, )</ID>
```

so after your change will the baseline create a simplified signature like this?
```xml
<ID>LongParameterList:TestClass.kt$TestClass</ID>
```
 and store all the violations that happens for this rule, doesn't matter if its 1, or more.
Sorry if I'm missing something obvious here

**Comment by BraisGabin** [comment]
My proposal doesn't fix your use case. @atulgpt proposal fixes it. But that proposal generates other problems. My proposal try to fix those other problems.

@atulgpt I think that we should go with just the name of the class for all the constructors.

**Comment by atulgpt** [comment]
Hi @BraisGabin so we are leaning towards the below changes(it might effect on other rules(like `CommentOverPrivateFunction`, `UndocumentedPublicFunction`, `CyclomaticComplexMethod` etc), as same rationale will be applicable for those rules). Right?

```kt
class A -> FileName.kt$A
class A(val a: Int, b: String) -> FileName.kt$A

constructor(a: Int, b: String): this() -> FileName.kt$A$ctor // different name instead of `ctor` we can decide

class A {
    fun foo(a: Int) -> FileName.kt$A$foo
    fun foo(a: Int, b: String) -> FileName.kt$A$foo
    fun Long.foo(a: Int, b: String) -> FileName.kt$A$foo
    // all functions with name foo and same for class level function
}
```

**Comment by BraisGabin** [comment]
Your first example should also contain the `FileName.kt`, right? Other than that I agree with your list.
